File: TodoTaskManagementUI/src/types/task.ts
export const TaskStatus = {
    Todo: 0,
    InProgress: 1,
    Done: 2,
} as const
export type TaskStatus = typeof TaskStatus[keyof typeof TaskStatus]

export interface TaskApi {
    id?: string
    title?: string | null
    description?: string | null
    dueDate?: string | null
    status?: TaskStatus | null
    isArchived?: boolean | null
    createdAt?: string | null
}

File: TodoTaskManagement/TodoTaskManagement.Domain/Tasks/Task.cs
namespace TodoTaskManagement.Domain.Tasks;

public enum TaskStatus
{
    Todo = 0,
    InProgress = 1,
    Done = 2
}

public class Task
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? DueDate { get; set; }
    public TaskStatus Status { get; set; } = TaskStatus.Todo;
    public bool IsArchived { get; set; } = false;
    public DateTime CreatedAt { get; set; }
}
